politeiawww/www.go

```diff
@@ -510,12 +511,29 @@ func _main() error {
 		if err != nil {
 			return fmt.Errorf("cmsdb setup: %v", err)
 		}
+
 		p.cron = cron.New()
 		p.checkInvoiceNotifications()
+
+		// XXX set up address watcher by checking db for approved invoices
+		// and get their addresses and add them to the watcher.
+
+		// XXX how many addresses should we plan on storing?
+		p.currentSubs = make([]string, 0, 1048)
+		p.addressWatcherChan = make(chan *addressWatcherContext)
+
+		p.wg = sync.WaitGroup{}
+		p.wg.Add(1)
+		go p.setupWatcher()
+		p.wg.Wait()
```

Comment:
If `p.wsClient` isn't nil, you'll want to do `p.wsclient.Close()` to shutdown the psclient loop and disconnect the websocket connection.

Reply:
Might need to run `p.setupWatcher()` synchronously though, and in `setupWatcher()` start the for loop in a goroutine after it's all setup.